presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -385,7 +389,18 @@ public CounterStat getHighMemorySplitSource()
             }
         }
 
-        Optional<HiveStorageFormat> storageFormat = getHiveStorageFormat(table.getStorage().getStorageFormat());
+        StorageFormat storageFormat = table.getStorage().getStorageFormat();
+        Optional<HiveStorageFormat> hiveStorageFormat = getHiveStorageFormat(storageFormat);
+
+        Optional<HiveStorageFormat> resolvedHiveStorageFormat;
```

Comment:
how about:
```
Optional<HiveStorageFormat> resolvedHiveStorageFormat = hiveStorageFormat;
if (isUseParquetColumnNames(session)) {
 ...
}
```

Reply:
@zhenxiao The variable `resolvedHiveStorageFormat` is being used in a lambda function so it has to be a `final` or an `effectively final` variable. If I set it before the if condition it won't remain an `effectively final` variable as it's value will change for the second time inside the if block. 

Due to this reason I had to use an else block.

Let me know if you think I can improve it some other way.

Reply:
I might miss something. which lambda function is used for `resolvedHiveStorageFormat`? seems it is only used in `getTableToPartitionMapping`?